zerver/models.py

```diff
@@ -3069,19 +3069,33 @@ def is_status_message(content: str, rendered_content: str) -> bool:
     class Meta:
         indexes = [
             GinIndex("search_tsvector", fastupdate=False, name="zerver_message_search_tsvector"),
-            models.Index(Upper("subject"), name="upper_subject_idx"),
-            models.Index("date_sent", name="zerver_message_date_sent_3b5b05d8"),
             models.Index(
-                "recipient",
+                "realm_id", "recipient_id", "id", name="zerver_message_realm_recipient_id"
+            ),
+            models.Index(
+                "realm_id",
+                "sender_id",
+                "recipient_id",
+                name="zerver_message_realm_sender_recipient",
+            ),
+            models.Index(
+                "realm_id", "sending_client_id", name="zerver_message_realm_sending_client"
+            ),
```

Comment:
What is this index used for?

Reply:
Ahh, probably this:

```
analytics/lib/counts.py:                UserCount, count_message_by_user_query(realm), (Message, "sending_client_id")
```

We should probably add a comment since it's kinda a single-purpose index.

Reply:
In practice, even with both of these in place, this actually ends up using the date index, so I dropped it from the latest revision.